### DISCUSSION TITLE
chaiscript+meta.hpp

### DISCUSSION THREAD

**Discussion by innochenti** [BlackMATov/meta.hpp#29] [General]
It's rather a question, not an issue. I was thinking about auto-binding reflected types to chaiscript, but can't figure out how to glue meta.hpp and chaiscript together.

To bind the type to chaiscript I need explicitly define all type's information: type, ctor, functions, etc.
For example:
chai.add(chaiscript::user_type<Vertex>(), "Vertex");
chai.add(chaiscript::constructor<Vertex(int32_t)>(), "Vertex");
chai.add(chaiscript::fun(&addVertex), "addVertex");

Is there a way to do it automatically using meta.hpp?

**Comment by BlackMATov** [comment]
I'm not familiar enough with the API of this library, but it seems that with dynamic reflection (like meta.hpp) this is not possible. Instead, for this API, we must use static reflection because we need to bind types that we need to know at compile time. In dynamic reflection, we don't have this knowledge at runtime.

P.S.
We can do automatic bindings with the help of such API that "Lua" has, for example. Because it doesn't require static types for bindings.

**Comment by innochenti** [comment]
I think it'd be great to have meta+lua sample
